src/main.rs

```diff
@@ -185,13 +194,9 @@ enum Bound {
     Date(Date<Utc>),
 }
 
-#[derive(Fail, Debug)]
-#[fail(display = "will never happen")]
-struct BoundParseError {}
-
 impl FromStr for Bound {
-    type Err = BoundParseError;
-    fn from_str(s: &str) -> Result<Bound, BoundParseError> {
+    type Err = NeverError;
```

Comment:
Is this a necessary error?  I renamed it to make it a more general error type, but I think that it can be eliminated altogether?